### PR TITLE
Refactors playing cards + fixes some issues with them

### LINKED ISSUE #53029: X4 doesn't work when planted on a card in a deck of cards
## Reproduction:
1. Buy X4 from your uplink (possibly also C4)
2. Set the X4 timer to 60000 seconds
3. Attach a signaler to one of the wires in the X4
4. Remove a playing card from a deck of playing cards.
5. Attach the X4 to the playing card.
6. Put the playing card back into the deck.
7. Activate the signaler.
8. Watch as the deck of playing cards utterly fails to explode.

This is probably a nullspace issue similar to #52247 and #52950

### PR DESCRIPTION
**Pull request by SuperNovaa41**
## About The Pull Request
Currently the code is poopy + cards are created and qdel'd upon adding to or removing from a deck.
That's fucking stupid. It causes stupid issues like https://github.com/tgstation/tgstation/issues/53029 and probably undocumented ones.
Fixes that so that the decks are lists of cards. I'm fairly certain this won't be a performance loss, I didn't have any issues on local (but thats a given), and lemon seems to think this would be fine too.
Also adds a bunch of autodoc comments to make this 9+ year old code a little better

## Why It's Good For The Game
bug fix + refactor
Fixes https://github.com/tgstation/tgstation/issues/53029

## Changelog

:cl:
refactor: Decks of cards are actual lists of cards now + other improvements.
fix: You can now put X4/C4 onto playing cards, and put them into decks without deleting your explosive.
/:cl:

### CODE REVIEW

**Comment by SuperNovaa41** [comment]
fucking whiteship

**Comment by timothymtorres** [comment]
Very nice!  Was planning to do this myself at some point once I got another computer.  

One thing I would like to point out is that there should be a possibility to remove cards from different decks and mix them together.  Due to this the top card of the deck should auto update the icon_state.

**Comment by SuperNovaa41** [comment]
> Very nice! Was planning to do this myself at some point once I got another computer.
> 
> One thing I would like to point out is that there should be a possibility to remove cards from different decks and mix them together. Due to this the top card of the deck should auto update the icon_state.

maybe in a different PR, i want to keep all the functionality the same

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -698,64 +697,139 @@
...
+
 /obj/item/toy/cards/deck
 	name = "deck of cards"
 	desc = "A deck of space-grade playing cards."
 	icon = 'icons/obj/toy.dmi'
 	deckstyle = "nanotrasen"
 	icon_state = "deck_nanotrasen_full"
 	w_class = WEIGHT_CLASS_SMALL
+	///Last time since shuffling the deck.
 	var/cooldown = 0
```
Comment (tralezab) [inline code/game/objects/items/toys.dm:794]: make this into our cooldown macros while you're here :)

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -698,64 +697,139 @@
...
+	///Cards in this deck
 	var/list/cards = list()
 
 /obj/item/toy/cards/deck/Initialize(mapload)
 	. = ..()
 	AddElement(/datum/element/drag_pickup)
 	populate_deck()
 
-///Generates all the cards within the deck.
+/obj/item/toy/cards/deck/proc/generate_card(name)
```
Comment (tralezab) [inline code/game/objects/items/toys.dm:805]: doc this proc!

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -894,15 +950,18 @@
 	return TRUE
 
 /**
+ * update_sprite
```
Comment (tralezab) [inline code/game/objects/items/toys.dm:953]: i love it when dmdocs have this, nice

**Review by tralezab — CHANGES_REQUESTED** [review]
small things, great job overall

**Review by Maurukas — APPROVED** [review]
Mapping is fine

**Pull request merged**